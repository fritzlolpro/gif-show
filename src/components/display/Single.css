.single-photo-display {
  padding-top: 48px;
}
.single-photo-display .single-photo {
  width: 40%;
  margin: 0 auto;
}
.single-photo-display .single-photo.thumbnail {
  width: 30%;
  vertical-align: top;
  display: inline-block;
}
.single-photo-display .single-photo.prev {
  float: left;
}
.single-photo-display .single-photo.next {
  float: right;
}
.single-photo-display .thumbnails {
  width: 40%;
  margin: 0 auto;
  height: 250px;
}
