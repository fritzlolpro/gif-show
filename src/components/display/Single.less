.single-photo-display {
    padding-top: 48px;
    .single-photo {
        width: 40%;
        margin: 0 auto;
        &.thumbnail {
            width: 30%;
            vertical-align: top;
            display: inline-block;
        }
        &.prev {
            float: left;
        }
        &.next {
            float: right;
        }
    }
    .thumbnails {
        width: 40%;
        margin: 0 auto;
        height: 250px;
    }
}