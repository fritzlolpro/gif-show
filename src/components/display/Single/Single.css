.single-photo-display {
  padding-top: 48px;
}
.single-photo-display .single-photo {
  width: 40%;
  margin: 0 auto;
}
@media only screen and (max-width : 479px) and (min-width : 321px), only screen and (max-width : 1023px) and (min-width : 480px) {
  .single-photo-display .single-photo {
    width: 90%;
  }
}
.single-photo-display .single-photo.thumbnail {
  width: 30%;
  vertical-align: top;
  display: inline-block;
}
.single-photo-display .single-photo.prev {
  float: left;
}
.single-photo-display .single-photo.next {
  float: right;
}
.single-photo-display .thumbnails {
  width: 40%;
  margin: 0 auto;
  height: 250px;
}
.single-photo-display .thumbnails > a {
  position: absolute;
  left: 50%;
  -webkit-transform: translate(-50%);
          transform: translate(-50%);
}
@media only screen and (max-width : 479px) and (min-width : 321px), only screen and (max-width : 1023px) and (min-width : 480px) {
  .single-photo-display .thumbnails {
    width: 90%;
  }
}
