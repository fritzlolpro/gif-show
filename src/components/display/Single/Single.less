@iphones: ~"only screen and (max-width : 479px) and (min-width : 321px)";
@tablet: ~"only screen and (max-width : 1023px) and (min-width : 480px)";
.single-photo-display {
    padding-top: 48px;
    .single-photo {
        width: 40%;
        margin: 0 auto;
        @media @iphones,
        @tablet {
            width: 90%;
        }
        &.thumbnail {
            width: 30%;
            vertical-align: top;
            display: inline-block;
        }
        &.prev {
            float: left;
        }
        &.next {
            float: right;
        }
        .error-output {
            margin-bottom: 40px;
        }
    }
    .thumbnails {
        width: 40%;
        margin: 0 auto;
        height: 250px;
        > a {
            position: absolute;
            left: 50%;
            transform: translate(-50%);
        }
        @media @iphones,
        @tablet {
            width: 90%;
        }
    }
}