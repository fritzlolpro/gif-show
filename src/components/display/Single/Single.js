import React from 'react'
import Closer from '../../control/Closer/Closer'
import Photo from '../Photo/Photo'
import './Single.css'
import {Link} from 'react-router-dom'

const Single = ({photos, activeId, onClick}) => {

    const selectedPhoto = photos.filter(photo => activeId === photo.id)[0] || false
    const photosIds = photos.map(photo => photo.id)
    const prevPhotoId = photosIds[photosIds.indexOf(activeId) - 1]
    const nextPhotoId = photosIds[photosIds.indexOf(activeId) + 1]
    const prevPhoto = photos.filter(photo => photo.id === prevPhotoId)[0]
    const nextPhoto = photos.filter(photo => photo.id === nextPhotoId)[0]

    return (
        <div className='single-photo-display'>

            <div className='single-photo'>
                {!!selectedPhoto
                    ? <Photo photo={selectedPhoto} clickable={false}/>
                    : <div className='error-output'>Error loading photo. <br/> <Link to='/'> To the main page </Link> </div>}
            </div>

            <div className='thumbnails'>
                {!!prevPhoto && <div onClick={onClick} className='single-photo thumbnail prev'>
                    <Photo photo={prevPhoto}/>
                </div>}

                <Closer/>

                {!!nextPhoto && <div onClick={onClick} className='single-photo thumbnail next'>
                    <Photo photo={nextPhoto}/>
                </div>}
            </div>
        </div>
    )
}
export default Single