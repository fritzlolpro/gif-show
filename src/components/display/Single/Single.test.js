import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Enzyme, {shallow, render} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({adapter: new Adapter()})

import Single from './Single'

describe('Single', () => {

    it('Defined', () => {
        expect(Single).toBeDefined();
    });

    it('Renders without props', () => {
        const tree = shallow(<Single/>);
        expect(tree).toMatchSnapshot();
    });

    const props = {
        photos: [
            {
                id: '123',
                title: 'test title',
                images: {
                    downsized_medium: {
                        url: 'test_url'
                    }
                }
            }
        ]
    }
    it('Renders with props', () => {
        const tree = shallow(<Single {...props}/>);
        expect(tree).toMatchSnapshot();
    });

})