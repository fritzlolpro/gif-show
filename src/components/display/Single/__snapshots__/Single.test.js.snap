// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Renders with props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Single
    photos={
        Array [
            Object {
              "id": "123",
              "images": Object {
                "downsized_medium": Object {
                  "url": "test_url",
                },
              },
              "title": "test title",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="single-photo"
>
          <div
                    className="error-output"
          >
                    Error loading photo.
                    <br />
                    <Link
                              replace={false}
                              to="/"
                    >
                              To the main page
                    </Link>
          </div>
</div>,
        <div
          className="thumbnails"
>
          <Closer />
           
          <div
                    className="single-photo thumbnail next"
                    onClick={undefined}
          >
                    <Photo
                              photo={
                                        Object {
                                                  "id": "123",
                                                  "images": Object {
                                                    "downsized_medium": Object {
                                                      "url": "test_url",
                                                    },
                                                  },
                                                  "title": "test title",
                                                }
                              }
                    />
          </div>
</div>,
      ],
      "className": "single-photo-display",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="error-output"
>
            Error loading photo.
            <br />
            <Link
                        replace={false}
                        to="/"
            >
                        To the main page
            </Link>
</div>,
          "className": "single-photo",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Error loading photo.",
              <br />,
              <Link
                replace={false}
                to="/"
>
                To the main page
</Link>,
            ],
            "className": "error-output",
          },
          "ref": null,
          "rendered": Array [
            "Error loading photo.",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "To the main page",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "To the main page",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <Closer />,
            " ",
            <div
              className="single-photo thumbnail next"
              onClick={undefined}
>
              <Photo
                            photo={
                                          Object {
                                                        "id": "123",
                                                        "images": Object {
                                                          "downsized_medium": Object {
                                                            "url": "test_url",
                                                          },
                                                        },
                                                        "title": "test title",
                                                      }
                            }
              />
</div>,
          ],
          "className": "thumbnails",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Photo
                photo={
                                Object {
                                                "id": "123",
                                                "images": Object {
                                                  "downsized_medium": Object {
                                                    "url": "test_url",
                                                  },
                                                },
                                                "title": "test title",
                                              }
                }
/>,
              "className": "single-photo thumbnail next",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "photo": Object {
                  "id": "123",
                  "images": Object {
                    "downsized_medium": Object {
                      "url": "test_url",
                    },
                  },
                  "title": "test title",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="single-photo"
>
            <div
                        className="error-output"
            >
                        Error loading photo.
                        <br />
                        <Link
                                    replace={false}
                                    to="/"
                        >
                                    To the main page
                        </Link>
            </div>
</div>,
          <div
            className="thumbnails"
>
            <Closer />
             
            <div
                        className="single-photo thumbnail next"
                        onClick={undefined}
            >
                        <Photo
                                    photo={
                                                Object {
                                                            "id": "123",
                                                            "images": Object {
                                                              "downsized_medium": Object {
                                                                "url": "test_url",
                                                              },
                                                            },
                                                            "title": "test title",
                                                          }
                                    }
                        />
            </div>
</div>,
        ],
        "className": "single-photo-display",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="error-output"
>
              Error loading photo.
              <br />
              <Link
                            replace={false}
                            to="/"
              >
                            To the main page
              </Link>
</div>,
            "className": "single-photo",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Error loading photo.",
                <br />,
                <Link
                  replace={false}
                  to="/"
>
                  To the main page
</Link>,
              ],
              "className": "error-output",
            },
            "ref": null,
            "rendered": Array [
              "Error loading photo.",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "To the main page",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "To the main page",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <Closer />,
              " ",
              <div
                className="single-photo thumbnail next"
                onClick={undefined}
>
                <Photo
                                photo={
                                                Object {
                                                                "id": "123",
                                                                "images": Object {
                                                                  "downsized_medium": Object {
                                                                    "url": "test_url",
                                                                  },
                                                                },
                                                                "title": "test title",
                                                              }
                                }
                />
</div>,
            ],
            "className": "thumbnails",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Photo
                  photo={
                                    Object {
                                                      "id": "123",
                                                      "images": Object {
                                                        "downsized_medium": Object {
                                                          "url": "test_url",
                                                        },
                                                      },
                                                      "title": "test title",
                                                    }
                  }
/>,
                "className": "single-photo thumbnail next",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "photo": Object {
                    "id": "123",
                    "images": Object {
                      "downsized_medium": Object {
                        "url": "test_url",
                      },
                    },
                    "title": "test title",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Single Renders without props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Single />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Error occured",
      "className": "error-output",
    },
    "ref": null,
    "rendered": "Error occured",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Error occured",
        "className": "error-output",
      },
      "ref": null,
      "rendered": "Error occured",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
